src/document.c

```diff
@@ -573,6 +575,15 @@ int Document_AddToIndexes(RSAddDocumentCtx *aCtx) {
 
       PreprocessorFunc pp = preprocessorMap[ii];
       if (pp(aCtx, &doc->fields[i], fs, fdata, &aCtx->status) != 0) {
+        if (!AddDocumentCtx_IsBlockable(aCtx)) {
+          ++aCtx->spec->stats.indexingFailures;
+        } else {
+          RedisModule_ThreadSafeContextLock(RSDummyContext);
+          if (aCtx->spec->uniqueId == aCtx->specId) {
```

Comment:
@ashtul the idea with checking the unique id is to prevent accessing spec which was free. So you can not really access the aCtx->spec without verifying the spec pointer is valid. you need to reopen the spec (using the spec name) and then compare the ids (or if you failed to open you know the spec was deleted).

Reply:
Done